backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -350,22 +350,32 @@ public RepositoryEntity getProtectedRepositoryById(
 
   private Optional<RepositoryEntity> getRepositoryByName(
       Session session, String name, Workspace workspace, RepositoryTypeEnum repositoryType) {
-    List<Long> repositoryIds = getRepositoryEntityIdsByName(session, name, repositoryType);
+    ModelDBServiceResourceTypes modelDBServiceResourceTypes =
+        ModelDBServiceResourceTypes.REPOSITORY;
+    if (repositoryType.equals(RepositoryTypeEnum.DATASET)) {
+      modelDBServiceResourceTypes = ModelDBServiceResourceTypes.DATASET;
+    }
     List<GetResourcesResponseItem> accessibleAllWorkspaceItems =
         roleService.getResourceItems(
-            workspace,
-            !repositoryIds.isEmpty()
-                ? repositoryIds.stream().map(String::valueOf).collect(Collectors.toSet())
-                : Collections.emptySet(),
-            ModelDBServiceResourceTypes.REPOSITORY);
-    Optional<Long> repoId =
+            workspace, Collections.emptySet(), modelDBServiceResourceTypes);
```

Comment:
Why do we list everything if we're getting a single element? It's wasteful. Don't we have the option to fetch by name? Or even fetching by ids subset like before.

Reply:
I have implemented overfetch because `getResource` with name is not implemented OR there is not have any other option to fetch resources based on the `name`.  Could you mine to implement it at UAC?